pandas/core/groupby.py

```diff
@@ -4419,7 +4419,9 @@ def __iter__(self):
             yield i, self._chop(sdata, slice(start, end))
 
     def _get_sorted_data(self):
-        return self.data.take(self.sort_idx, axis=self.axis, convert=False)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            return self.data.take(self.sort_idx, axis=self.axis, convert=False)
```

Comment:
@jreback @jorisvandenbossche : This is a tricky one for me here.  On the one hand, I want to remove `convert=False`, but because `axis` of the `groupby` object might be invalid when calling `Series.take`, this is not possible (when `True`, it re-indexes the value according to the length of the slice along the index).

On the other hand, this `catch_warnings()`, while an easy way to get things to pass for now, is not a solution I want to merge.  Suggestions on how to handle this problem?

Reply:
Actually, might have found the answer myself 😄